### PR TITLE
Anonymize config output

### LINKED ISSUE #20: `viash.yaml` contains absolute paths from build host
The `viash.yaml` file written as part of a `build` contains absolute paths from the system where the build was performed. These should be made relative so there is no trace to the developer/build system.

### PR DESCRIPTION
**Pull request by Grifs**
## Describe your changes

Anonymize configs before outputting them, either to console or file

Add .config.projectDir so we can always access it without always having to pass the path as an argument to functions
Set .config.projectDir to @internalFunctionality so it's never outputted (would defeat the whole purpose)
This could potentially clean up more code, but this is not done yet

What if there is no projectDir set (no viash project file found). We'll still be outputting full paths!

## Related issue(s)

Closes #20 

## Type of Change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New functionality (non-breaking change which adds functionality)
- [ ] Major change (non-breaking change which modifies existing functionality)
- [ ] Minor change (non-breaking change which does not modify existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

## Checklist

Requirements:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc.
- [x] I have performed a self-review of my code by checking the "Changed Files" tab.
- [x] My code follows the code style of this project.

Tests:

- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

Documentation:

- [x] Proposed changes are described in the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by rcannood — CHANGES_REQUESTED** [review]

File: src/main/scala/io/viash/config/Config.scala [hunk 1]
```diff
@@ -87,7 +87,10 @@ case class Config(
   engines: List[Engine] = Nil,
 
   @undocumented
-  info: Option[Info] = None
+  info: Option[Info] = None,
+
+  @internalFunctionality
+  projectDir: Option[String] = None,
```
[the remark this answers is not in the document]
Comment (Grifs) [reply src/main/scala/io/viash/config/Config.scala]: Absolutely! The base didn't have the ProjectConfig in the (component) config yet. Fixed by merging develop into the branch (together with some other functionality fixes)

File: src/test/scala/io/viash/helpers/IOTest.scala [hunk 2]
```diff
@@ -94,4 +95,39 @@ class IOTest extends AnyFunSuite with BeforeAndAfter {
...
+    val result = IO.anonymizePath(basePath, path)
+    assert(result == "[anonymized]/test.txt")
+  }
+
+  test("anonymizePath with a common path and a trailing slash") {
+    val basePath = Some(Paths.get("/foo/bar"))
+    val path = "/foo/bar/baz/test.txt"
+    val result = IO.anonymizePath(basePath, path)
+    assert(result == "baz/test.txt")
+  }
```
Comment (rcannood) [inline src/test/scala/io/viash/helpers/IOTest.scala:125]: There is no trailing slash in this example, or am I mistaken?

```suggestion
  test("anonymizePath with a common path and a trailing slash") {
    val basePath = Some(Paths.get("/foo/bar/"))
    val path = "/foo/bar/baz/test.txt"
    val result = IO.anonymizePath(basePath, path)
    assert(result == "baz/test.txt")
  }
```

Or


```suggestion
  test("anonymizePath with a common path and a trailing slash") {
    val basePath = Some(Paths.get("/foo/bar"))
    val path = "/foo/bar/baz/test_folder/"
    val result = IO.anonymizePath(basePath, path)
    assert(result == "baz/test_folder/")
  }
```

**Review by rcannood — APPROVED** [review]
Suggestions LGTM!

Minor question about the test :)

File: src/test/scala/io/viash/helpers/IOTest.scala (hunk 2 above)
Comment (Grifs) [reply src/test/scala/io/viash/helpers/IOTest.scala:125]: Thanks for spotting that! A refactor went wrong. :stuck_out_tongue:

**Pull request merged**